capstone/src/main/java/com/google/sps/Objects/Comment.java

```diff
@@ -22,6 +24,34 @@ public static Comment getCommentEntity(EmbeddedEntity entity) {
     return new Comment(timestamp, commentText, userId);
   }
 
+  public static EmbeddedEntity toEntity(String commentText, String userId) {
+    EmbeddedEntity commentEntity = new EmbeddedEntity();
+    commentEntity.setProperty("timestamp", System.currentTimeMillis());
+    commentEntity.setProperty("commentText", commentText);
+    commentEntity.setProperty("userId", userId);
+    return commentEntity;
+  }
+
+  public static ArrayList<EmbeddedEntity> createCommentEntities(ArrayList<Comment> comments) {
+    ArrayList<EmbeddedEntity> allComments = new ArrayList<>();
+    for (Comment comment: comments) {
+      allComments.add(
+        toEntity(comment.getCommentText(), comment.getUser())
+      );
+    } 
+    return allComments;
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    if (other == null) return false;
+    if (other == this) return true;
+    if (!(other instanceof Comment)) return false;
+    Comment comment = (Comment) other;
+    return commentText.equals(comment.commentText) &&
+      userId.equals(comment.userId);
```

Comment:
I agree that my current approach is simply checking if the comment value is the expected value, but doesn't check for an unique identifier (timestamp/commentId). Since the comment is an embedded entity, I can't generate an Id for the comment. 
Datastore allows embedded entities to have keys: "You can optionally associate a key with an embedded entity, but (unlike a full-fledged entity) the key is not required and, even if present, cannot be used to retrieve the entity." However, since I can't retrieve a comment by key this may pose a problem for editing comments, so we may need to change this to a regular entity. 

I'll keep this in mind, but will leave the equals check as is for now.